### PR TITLE
Show event documents

### LINKED ISSUE #401: Show documents in the events page
### What
Some events have documents, like pool party, these are not shown in the app

### Why
because people really need to read these important documents

### PR DESCRIPTION
**Pull request by Scarletto**
Closes #401.

### Summary
Made class `Document` (as a `JsonSerializable`) and added it to the `Event` model and serialization.

Added dependency `open_file_plus`, a helpful small package that makes opening downloaded files in external apps easy and clean.

Made `FileButton`, an `ElevatedButton` that gets a file via `ThaliaCacheManager()` and opens it in an external application.

Added `FileButton`s for all event documents in `EventScreen`.

### How to test
Steps to test the changes you made:
1. Go to any event with documents like TPay Test.
2. Click on any document.

### CODE REVIEW

**Comment by Scarletto** [comment]
hella old dart version ayo

File: lib/utilities/filetype_translator.dart [hunk 1]
```diff
@@ -0,0 +1,201 @@
+String? extensionToType(String extension) {
+  switch (extension) {
```
Comment (JeeVee11) [inline lib/utilities/filetype_translator.dart:2]: Please use a library

**Review by JeeVee11 — DISMISSED** [review]

File: lib/utilities/filetype_translator.dart (hunk 1 above)
Comment (Scarletto) [reply lib/utilities/filetype_translator.dart]: I fixed it

**Comment by JAicewizard** [comment]
Did it work out removing the mime-type converters?

**Comment by Scarletto** [comment]
> Did it work out removing the mime-type converters?

Yes, I now download the file, copy the file with the proper extension and put it in the cache, delete the old cached file.

**Review by JAicewizard — APPROVED** [review]

**Comment by JAicewizard** [comment]
You can merge if you want. Would be nice if we could not use the file extension and just base it off MIME but I dont think this will be abused

**Pull request merged**